src/com/hms/flexyosisoftconnector/DataPoster.java

```diff
@@ -23,25 +35,56 @@ public void run()
          for (int tagIndex = 0; tagIndex < OSIsoftConfig.tags.size(); tagIndex++) {
             dataPoints.add(((Tag) OSIsoftConfig.tags.get(tagIndex)).getNewestDataPoints(pointsPerTag));
 
+            String tagName = ((Tag) OSIsoftConfig.tags.get(tagIndex)).getTagName();
+            if(omfEnabled) {
+               if (tagIndex > 0 ) {
+                  OSIsoftServer.seperateDataMessage();
+               }
+
+               OSIsoftServer.addContainerStartToOMFdataMessage(tagName);
+            }
+
             for(int dataPointIndex = 0; dataPointIndex<((ArrayList) dataPoints.get(tagIndex)).size(); dataPointIndex++)
             {
+
                if(((ArrayList)dataPoints.get(tagIndex)).get(dataPointIndex) != null)
                {
-                  OSIsoftServer.addPointToBatch(((Tag) OSIsoftConfig.tags.get(tagIndex)),(DataPoint)((ArrayList)dataPoints.get(tagIndex)).get(dataPointIndex));
+
+                  if(omfEnabled) {
+                     String tagValue = ((Tag) OSIsoftConfig.tags.get(tagIndex)).getTagValue();
+                     String timeStamp = ((DataPoint)((ArrayList)dataPoints.get(tagIndex)).get(dataPointIndex)).timestamp;
+
+                     // if there is more than one data point we need to comma separate them
+                     if (dataPointIndex > 0) {
+                        OSIsoftServer.seperateDataMessage();
+                     }
+
+                     // add a data point to the OMF message body
+                     OSIsoftServer.addPointToOMFdataMessage(tagValue, timeStamp);
+                  } else {
+                     OSIsoftServer.addPointToBatch(((Tag) OSIsoftConfig.tags.get(tagIndex)),(DataPoint)((ArrayList)dataPoints.get(tagIndex)).get(dataPointIndex));
+                  }
                   pointsAdded++;
                }
                else Logger.LOG_ERR("Null data point encountered");
             }
+            if(omfEnabled) {
+               OSIsoftServer.addContainerEndToOMFdataMessage();
+            }
+         }
 
+         if(omfEnabled) {
+            OSIsoftServer.endOMFdataMessage();
+         } else {
+            OSIsoftServer.endBatch();
          }
 
-         OSIsoftServer.endBatch();
 
          if(pointsAdded>0)
          {
             Logger.LOG_DEBUG("Sending " + pointsAdded + " points to server");
             //Post the tags to the server
-            boolean retval = OSIsoftServer.postBatch();
+            boolean retval = OSIsoftServer.postOMFBatch();

```

Comment:
Related to my last comment, wouldn't this break web api functionality?

Reply:
it is moved to line 28

Reply:
wait a second, no it isn't

Reply:
oh, maybe the old stuff is not working. it is posting the old api message body to the omf url. So I probably do not have OMF disabled. It's weird that that even works. I might not be able to disable it without uninstalling piwebapi and getting an older version

Reply:
fixed this, still need to test